### PR TITLE
[Feature][Dataobject] add direct upload for video

### LINKED ISSUE #13321: [Feature Request]: Direct Asset Upload for Video
### Feature description

In Pimcore we have the possibility to open up the direct asset upload dialog in the many-to-one relation component (if assets are allowed in the configuration of the component) and also we can do the same in all the media components regarding images like "Image", Image Advanced" and "Image Gallery".

On video components, once we select the "attest" video type, there's only the possibility to attach an existing asset. The option of directly uploading an asset is missing.

I think that this kind of feature could be beneficial for users, in terms of time-saving and ease of usage.

### PR DESCRIPTION
**Pull request by dpfahlbusch**
## Changes in this pull request  
Resolves #13321

### CODE REVIEW

**Comment by mcop1** [comment]
@dpfahlbusch could you please merge 11.x into your pr again (to fix the tests/phpstan)? Thanks!

File: models/DataObject/ClassDefinition/Data/Video.php [hunk 1]
```diff
@@ -152,6 +157,23 @@ public function setHeight($height)
         return $this;
     }
 
+    /**
+     * @param string $uploadPath
+     *
+     * @return $this
+     */
+    public function setUploadPath($uploadPath): static
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Video.php]:
```suggestion
    /**
     * @return $this
     */
    public function setUploadPath(string $uploadPath): static
```

Comment (dpfahlbusch) [reply models/DataObject/ClassDefinition/Data/Video.php]: Thanks @blankse

**Comment by dpfahlbusch** [comment]
@mcop1 pr is ready to be reviewed :)

**Comment by mcop1** [comment]
@dpfahlbusch Thanks, pr looks good to me, but I am having some problems with the toolbar layout.

With pimcore 11.x the video control looks like this:
grafik

However, with the changes from your pr, the button gets pushed to the right due to the icons on the left:
grafik

Maybe we could add the same functionality here as for the image control:
grafik

What do you think?

**Comment by dpfahlbusch** [comment]
It's a good idea. I will implement it :-)

**Comment by dpfahlbusch** [comment]
@mcop1 done :)

File: bundles/AdminBundle/public/js/pimcore/object/tags/video.js [hunk 2]
```diff
@@ -63,31 +63,117 @@ pimcore.object.tags.video = Class.create(pimcore.object.tags.abstract, {
...
-                iconCls: "pimcore_icon_video pimcore_icon_overlay_edit",
-                handler: this.openEdit.bind(this)
-            }, {
-                xtype: "button",
-                iconCls: "pimcore_icon_delete",
-                handler: this.empty.bind(this)
-            }],
+            tbar: {
+                overflowHandler: 'menu',
+                toolbarItems
```
Comment (mcop1) [inline bundles/AdminBundle/public/js/pimcore/object/tags/video.js]:
```suggestion
                items: toolbarItems
```

File: bundles/AdminBundle/public/js/pimcore/object/tags/video.js [hunk 3]
```diff
@@ -63,31 +63,117 @@ pimcore.object.tags.video = Class.create(pimcore.object.tags.abstract, {
...
+                    width: 48,
+                    height: 24,
+                    cls: "pimcore_icon_droptarget_upload"
+                }
+            )
+        }
+
+        toolbarItems.push({
+            xtype: "tbtext",
+            text: "<b>" + this.fieldConfig.title + "</b>"
```
Comment (mcop1) [inline bundles/AdminBundle/public/js/pimcore/object/tags/video.js]: We should add an overflow text for this toolbar item too

Comment (dpfahlbusch) [reply bundles/AdminBundle/public/js/pimcore/object/tags/video.js]: Done :)

File: bundles/AdminBundle/public/js/pimcore/object/tags/video.js [hunk 4]
```diff
@@ -63,31 +63,118 @@ pimcore.object.tags.video = Class.create(pimcore.object.tags.abstract, {
...
+        }
+
+        toolbarItems.push({
+            xtype: "tbtext",
+            text: "<b>" + this.fieldConfig.title + "</b>",
+            overflowText: this.fieldConfig.title
+        }, "->", {
+            xtype: "button",
+            iconCls: "pimcore_icon_video pimcore_icon_overlay_edit",
+            handler: this.openEdit.bind(this)
```
Comment (mcop1) [inline bundles/AdminBundle/public/js/pimcore/object/tags/video.js]:
```suggestion
            xtype: "tbtext",
            text: "<b>" + this.fieldConfig.title + "</b>",
        }, "->", {
            xtype: "button",
            overflowText: this.fieldConfig.title,
            iconCls: "pimcore_icon_video pimcore_icon_overlay_edit",
            handler: this.openEdit.bind(this)
```

**Comment by mcop1** [comment]
Thank you for your efforts and contribution, looks good to me now 👍

**Review by mcop1 — APPROVED** [review]

**Pull request merged**